### DISCUSSION TITLE
Parameter Scaling for optimization and simulation.

### DISCUSSION THREAD

**Discussion by lmriccardo** [PEtab-dev/PEtab#567] [Q&A] [answered]
I everyone!!

In the PEtab documentation it says: "parameterScale is the scale on which parameters are estimated". 

For what I have understood, prior to the optimization process: I sample the starting point (i.e., the values for the free parameters in the parameter table), then I scale those values using the corresponding scaling transformation (lin|log|log10) and then I start the optimization. Let's call these values $\theta$

Now, at each optimization step I need to compute the observables which, in turn, requires the simulation of the model up to a specific horizon. The parameters of the model, that needs to be estimated, are initialized with $\theta$, which is updated at each optimization step. 

The question is: **before starting the simulation, do I have to *reverse* scale $\theta$ or not?**

For example, let's assume that I have a parameter $p$ with `estimate=1`, bounds $[1.0E-5, 1000]$ and scaling $\log_{10}$. I sample a random value using the Latin Hypercube Sampling method and I obtain a value $v$ which then become $v' = \log_{10}(v)$, since I have to scale that value for the optimization. If $v < 1$, I obtain that $v' < 0$ and then, when I need to simulate, I set the corresponding model parameter $p$ to $v'$. If I look at the bounds, setting $p = v'$ is not correct (it is a negative value), hence I have to reverse the scaling and set $p = 10^{v'} = v$. 

Is this correct or I have to use $v'$ when simulating? It is not clear to me. 

Thanks for the help.

**Comment by dilpath** [comment, marked as answer]
Hi! Yes, you're correct, it should be the unscaled values for simulation.

It sounds like you're implementing PEtab for your workflow. If you are using the `libpetab-python` package, then there are helper methods there that you might find useful.

- `Problem.(un)scale_parameters`: https://github.com/PEtab-dev/libpetab-python/blob/52d8ab190765ca543ea8978368392658e94b9789/petab/problem.py#L845-L891
- `(un)scale` and `map_(un)scale`: https://petab.readthedocs.io/projects/libpetab-python/en/latest/build/_autosummary/petab.parameters.html
- various parameter mapping functions to connect simulation parameters to optimization parameters: https://petab.readthedocs.io/projects/libpetab-python/en/latest/build/_autosummary/petab.parameter_mapping.html

**Reply by lmriccardo** [reply]
Perfect, thank you for the clarification. You have been very helpfull!!!
